### ISSUE TITLE
Investigate the necessity of `nameOverride`

### ISSUE THREAD

**Issue by disrupted** [bakdata/streams-bootstrap#185]
Having to set the `nameOverride` for every deployment seems like a workaround. We should look into setting the correct Helm release name automatically without overriding.

**Comment by raminqaf** [comment]
Let's step back and see the differences between `nameOverride` and `fullnameOverride`. By default, whenever you create a Helm chart, the template file defines `nameOverride` and `fullnameOverride` like this:

```go
{{- define "<CHARTNAME>.fullname" -}}
{{- if .Values.fullnameOverride }}
{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
{{- else }}
{{- $name := default .Chart.Name .Values.nameOverride }}
{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
{{- end }}
{{- end }}
```
The template above can be broken down as follows:
- If no `nameOverride` and `fullNameOverride` is set -> Deployment name = `release-name-<chart-name`
- if `nameOverride` is set -> Deployment name = `release-name-<name-override>`
- if `fullnameOverride` is set -> Deployment name = `<full-name-override>`
However, in streams bootstrap, we have altered this code in [this commit](https://github.com/bakdata/streams-bootstrap/commit/a2a013cc99d216da07fa3fc00f19de46b790686b). Now the behavior of `nameOverride` and `fullnameOverride` is the same.
Maybe we should consider changing this parameter to `kafka-app-name`, which is not Helm-specific or keep either `nameOverride` or `fullnameOverride`.

**Comment by philipp94831** [comment]
Can this be closed? See discussion on #193
